storage/src/vespa/storage/bucketdb/bucketmanager.cpp

```diff
@@ -259,26 +259,40 @@ void BucketManager::updateMinUsedBits()
     }
 }
 
+namespace {
+
+void copyBucketInfoMap(const BucketManager::BucketInfoMap &src,
+                       BucketManager::BucketInfoMap &dst)
+{
+    for (const auto &srcElem : src) {
+        auto &dstList = dst[srcElem.first];
+        dstList.insert(dstList.end(), srcElem.second.begin(), srcElem.second.end());
+    }
+}
+
+}
+
 // Responsible for sending on messages that was previously queued
 void BucketManager::run(framework::ThreadHandle& thread)
 {
     const int64_t CHECK_MINUSEDBITS_INTERVAL = 1000*30;
     framework::MilliSecTime timeToCheckMinUsedBits(0);
     while (!thread.interrupted()) {
         bool didWork = false;
-        BIList infoReqs;
+        BucketInfoMap infoReqs;
         {
             vespalib::MonitorGuard monitor(_workerMonitor);
             infoReqs.swap(_bucketInfoRequests);
         }
 
-        didWork |= processRequestBucketInfoCommands(infoReqs);
+        for (auto &req : infoReqs) {
+            didWork |= processRequestBucketInfoCommands(req.first, req.second);
+        }
 
         {
             vespalib::MonitorGuard monitor(_workerMonitor);
             if (!infoReqs.empty()) {
-                infoReqs.insert(infoReqs.end(),
-                        _bucketInfoRequests.begin(), _bucketInfoRequests.end());
+                copyBucketInfoMap(_bucketInfoRequests, infoReqs);
```

Comment:
After some discussion, we have concluded that this code should never be reached. Replace with an assertion that all reqs have been cleared per bucket space instead.